modular_splurt/code/datums/components/pregnancy.dm

```diff
@@ -0,0 +1,426 @@
+	UnregisterSignal(carrier, COMSIG_HEALTH_SCAN)
+	UnregisterSignal(carrier, COMSIG_MOB_APPLY_DAMAGE)
+	UnregisterSignal(carrier, COMSIG_MOB_CLIMAX)
+	if(container)
+		UnregisterSignal(container, COMSIG_ORGAN_REMOVED)
+
+/datum/component/pregnancy/Destroy()
+	if(carrier)
+		generic_pragency_end()
+	return ..()
+
+/datum/component/pregnancy/PreTransfer()
+	if(carrier)
+		generic_pragency_end()
+	UnregisterFromParent()
+	oviposition = FALSE
+	carrier = null
+	container = null
+	egg_name = null
+
+/datum/component/pregnancy/PostTransfer()
+	if(isliving(parent))
+		carrier = parent
+		RegisterWithParent()
+	else if(isitem(parent))
+		max_stage += 1
+		RegisterWithParent()
+	else
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/pregnancy/proc/name_egg(datum/source, name)
+	SIGNAL_HANDLER
+
+	egg_name = name
+
+/datum/component/pregnancy/proc/on_climax(datum/source, atom/target, obj/item/organ/genital/sender, obj/item/organ/genital/receiver, spill)
+	SIGNAL_HANDLER
+
+	if(stage < 2)
+		return FALSE
+
+	to_chat(carrier, span_userlove("You feel your egg sliding out slowly inside!"))
+
+	if(receiver && isliving(target))
+		if(CHECK_BITFIELD(receiver.genital_flags, GENITAL_CAN_STUFF))
+			return lay_eg(receiver)
+	return lay_eg(get_turf(carrier))
+
+/datum/component/pregnancy/proc/on_obj_break(datum/source, damage_flag)
+	SIGNAL_HANDLER
+
+	qdel(src)
+
+/datum/component/pregnancy/proc/on_entering(datum/source, atom/destination, atom/oldLoc)
+	SIGNAL_HANDLER
+
+	if(istype(destination, /obj/item/organ))
+		var/obj/item/organ/preg_container = destination
+		//severed wombs don't count, boyos
+		if(!preg_container.owner)
+			return
+		carrier = preg_container.owner
+		container = preg_container
+		pregnancy_inflation = carrier.client?.prefs?.pregnancy_inflation
+		register_carrier()
+		generic_pragency_start()
+	else if(carrier)
+		generic_pragency_end()
+		unregister_carrier()
+		carrier = null
+		container = null
+
+/datum/component/pregnancy/proc/handle_life(seconds)
+	SIGNAL_HANDLER
+
+	if(ishuman(carrier) && pregnancy_inflation)
+		handle_belly_stuff(carrier)
+
+	if(oviposition)
+		handle_ovi_preg()
+	else
+		handle_incubation()
+
+	if((stage > (max_stage / 2)) && !revealed)
+		revealed = TRUE
+		carrier.apply_status_effect(/datum/status_effect/pregnancy)
+
+	if(stage > 3 && ishuman(carrier))
+		SEND_SIGNAL(carrier, COMSIG_ADD_MOOD_EVENT, "pregnancy", /datum/mood_event/pregnant_negative)
+
+	if(COOLDOWN_FINISHED(src, stage_time))
+		stage += 1
+		stage = min(stage, max_stage)
+		COOLDOWN_START(src, stage_time, PREGNANCY_STAGE_DURATION)
+
+/datum/component/pregnancy/proc/handle_belly_stuff(mob/living/carbon/human/gregnant)
+	if(stage < 2)
+		return
+	var/obj/item/organ/genital/belly/belly = gregnant.getorganslot(ORGAN_SLOT_BELLY)
+	if(belly)
+		switch(stage)
+			if(2)
+				belly.size = max(belly.size, 2)
+			if(3)
+				belly.size = max(belly.size, 3)
+			if(4)
+				belly.size = max(belly.size, 4)
+		belly.update()
```

Comment:
```suggestion
	if(!belly)
		return
	belly_growth += 1
	belly.modify_size(1)
```
The current way bellies are handled has several problems. When you lay the egg out and then stuff it back inside someone it'll spam belly size changing messages, which means it's constantly trying to change the belly size for some reason. It weirdly stops and makes the belly bigger for whatever reason after the egg has developed another step. If you again shove it into someone else however, their belly won't grow, and when you shove it back into yourself you'll get the spammed messages again. It'll also just shrink your belly to size zero in case it works properly

Besides, it's better to use the function made to actually handle belly sizes in order for all it's values to be properly updated.
![unknown](https://user-images.githubusercontent.com/93835010/177376160-019f360c-4e13-4277-9511-e5e255b55226.png)